test/unit/transitions/multi_form_alerts.js

```diff
@@ -0,0 +1,384 @@
+
+  transition.onMatch({ doc: doc }, undefined, undefined, () => {
+    const countedReports = vm.runInNewContext.getCalls().map((call) => call.args[1].report);
+    test.equals(countedReports.length, 2);
+    test.equals(countedReports[0]._id, doc._id);
+    test.equals(countedReports[1]._id, reports[0]._id);
+    test.done();
+  });
+};
+
+exports['evals isReportCounted'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  transition.onMatch({ doc: doc }, undefined, undefined, () => {
+    const script = vm.runInNewContext.args[0][0];
+    test.ok(script.includes(alert.isReportCounted));
+    test.done();
+  });
+};
+
+exports['if not enough reports pass the isReportCounted func, does nothing'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  // make all reports fail
+  vm.runInNewContext.restore();
+  sinon.stub(vm, 'runInNewContext').returns(false);
+
+  transition.onMatch({ doc: doc }, undefined, undefined, () => {
+    test.equals(vm.runInNewContext.getCalls().length, alert.numReportsThreshold); // 3
+    test.equals(messages.addError.getCalls().length, 0);
+    test.equals(messages.addMessage.getCalls().length, 0);
+    test.done();
+  });
+};
+
+exports['if no reports in time window, does nothing'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  utils.getReportsWithinTimeWindow.restore();
+  sinon.stub(utils, 'getReportsWithinTimeWindow').returns(Promise.resolve([]));
+
+  transition.onMatch({ doc: doc }, undefined, undefined, (err, docNeedsSaving) => {
+    test.equals(messages.addError.getCalls().length, 0);
+    test.equals(messages.addMessage.getCalls().length, 0);
+    test.ok(!docNeedsSaving);
+    test.done();
+  });
+};
+
+const assertMessage = (test, call, recipient, message) => {
+    const messageArgs = call.args[0];
+    test.deepEqual(messageArgs, {
+      doc: doc,
+      phone: recipient,
+      message: message,
+      options: { countedReports: [doc, ...hydratedReports] }
+    });
+};
+
+const assertMessages = (test, addMessageStub, alert) => {
+  addMessageStub.getCalls().forEach((call, i) => {
+    assertMessage(test, call, alert.recipients[i], alert.message);
+  });
+};
+
+exports['if enough reports pass the isReportCounted func, adds message'] = test => {
+  sinon.stub(config, 'get').returns([alert]);
+
+  transition.onMatch({ doc: doc }, undefined, undefined, (err, docNeedsSaving) => {
+    test.equals(vm.runInNewContext.getCalls().length, alert.numReportsThreshold); // 3
+    test.equals(messages.addError.getCalls().length, 0);
+
+    test.equals(messages.addMessage.getCalls().length, alert.recipients.length);
+    assertMessages(test, messages.addMessage, alert);
+
+    test.ok(!err);
+    test.ok(docNeedsSaving);
+    test.done();
+  });
+};
+
+exports['adds message when recipient is evaled'] = test => {
+  const recipient = 'countedReports[0].contact.phone';
+  sinon.stub(config, 'get').returns(
+    [_.defaults({ recipients: [recipient]} , alert)]);
```

Comment:
You use _.defaults a lot and it's quite hard to read. How about resetting `alert` to the default in the setup, and then just saying `alert.recipients = [recipient]` in the test? That is much more readable to me...